### DISCUSSION TITLE
get the image resolution (in dpi)

### DISCUSSION THREAD

**Discussion by pr-apes** [qpdf/qpdf#1045] [Q&A] [answered]
@jberkenbilt,

I would like to know the image resolution (in dpi).

Given the image data (`width`, in this case):

``` json
  "images": [
	{
	  "bitspercomponent": 8,
	  "colorspace": "/DeviceRGB",
	  "decodeparms": [
		null
	  ],
	  "filter": [
		"/DCTDecode"
	  ],
	  "filterable": false,
	  "height": 354,
	  "name": "/Im1",
	  "object": "1 0 R",
	  "width": 531
	}
  ],
```

And the media box dimensions:

``` json
  "obj:5 0 R": {
	"value": [
	  0,
	  0,
	  595.275590072,
	  841.88975789
	]
  },
```

All I would have to do is `image_width / (mediabox_width / 72)`.

This gives me about 65dpi. But I'm cheating and I know that the image resolution is 150dpi.

The above division only works if the image width is the full page width.

How do I know which is the image frame that would allow to get the image resolution (both displayed and real one)?

Many thanks for your help.

**Comment by jberkenbilt** [comment, marked as answer]
Sorry for the delay -- I overlooked this message. Knowing the image resolution requires following all the different places that can introduce transformation matrices. It doesn't require full page rendering, but it requires a lot of work. There is no way to know the image resolution from an image in isolation or image and page metadata. You can know the pixel dimensions of image, but to know the resolution, you have to be able to get the size of the box in which the image is displayed, and that requires parsing the content stream.
